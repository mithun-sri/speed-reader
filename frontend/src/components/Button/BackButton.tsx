import { IconButton } from "@mui/material";
import { useEffect, useState } from "react";

const BackButton: React.FC<{
  label?: string;
  handleClick?: () => void;
}> = ({ label, handleClick }) => {
  const [fontSize, setFontSize] = useState(calculateFontSize());

  useEffect(() => {
    function handleResize() {
      setFontSize(calculateFontSize());
    }
    window.addEventListener("resize", handleResize);
    return () => {
      window.removeEventListener("resize", handleResize);
    };
  }, []);

  function calculateFontSize() {
    const windowWidth = window.innerWidth;
    const minSize = 40;
    const maxSize = 180;
    const calculatedSize = Math.min(
      maxSize,
      Math.max(minSize, windowWidth / 6),
    );
    return calculatedSize;
  }

  const buttonText: string = label || "back";

  return (
    <IconButton
      onClick={() => {
        handleClick ? handleClick() : () => {};
      }}
      sx={{
        height: fontSize / 3.6,
        fontFamily: "JetBrains Mono, monospace",
        color: "#D1D0C5",
        fontWeight: "bolder",
        fontSize: fontSize / 6.8,
      }}
    >
      {`<_${buttonText}`}
    </IconButton>
  );
};

export default BackButton;
