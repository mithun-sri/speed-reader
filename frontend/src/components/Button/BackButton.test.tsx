import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import BackButton from "./BackButton";

it("Matches snapshot before/after click", () => {
  const component = renderer.create(<BackButton />);
  const tree1 = component.toJSON() as ReactTestRendererJSON;
  expect(tree1).toMatchSnapshot();

  renderer.act(() => {
    tree1.props.onClick();
  });
  const tree2 = component.toJSON() as ReactTestRendererJSON;
  expect(tree2).toMatchSnapshot();
});
