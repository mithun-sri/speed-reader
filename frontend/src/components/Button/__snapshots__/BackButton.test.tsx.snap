// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot before/after click 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-9kbnz4-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  &lt; back
</button>
`;

exports[`Matches snapshot before/after click 2`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-9kbnz4-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  &lt; back
</button>
`;
